src/main/java/carpet/script/api/Inventories.java

```diff
@@ -119,14 +121,7 @@ public static void apply(Expression expression)
                 }
             }
             List<Recipe<?>> recipes;
-            try
-            {
-                recipes = ((RecipeManagerInterface) cc.s.getMinecraftServer().getRecipeManager()).getAllMatching(type, new Identifier(recipeName));
-            }
-            catch (InvalidIdentifierException ignored)
-            {
-                return LazyValue.NULL;
-            }
+            recipes = ((RecipeManagerInterface) cc.s.getMinecraftServer().getRecipeManager()).getAllMatching(type, new Identifier(recipeName));
```

Comment:
not thrown here?

Reply:
At least not `InvalidIdentifierException`, since that one is for "syntax" errors when creating the `Identifier` instance, not the specific identifier not being present in there (eg having two colons or some other disallowed character).

It returns an empty list if none match (it is a custom implementation `RecipeManager_scarpetMixin`).

Now thinking, maybe `InvalidIdentifierException` would be good to be caught and be rethrown as an `InternalExpressionException`/`ExpressionException` every time when creating an `Identifier`, but it can happen basically anywhere, so it would be a bit tedious to add it to each and every function. Maybe could be added to `handleCodeException()`, like arithmetics.